persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideActor.scala

```diff
@@ -81,12 +81,15 @@ private[lagom] class ReadSideActor[Event <: AggregateEvent[Event]](
         ) { () =>
           val handler: ReadSideProcessor.ReadSideHandler[Event] = processorFactory().buildHandler()
           val futureOffset = handler.prepare(tag).toScala
-          scaladsl.Source.fromFuture(futureOffset).flatMapConcat {
-            offset =>
-              val eventStreamSource = eventStreamFactory(tag, offset).asScala
-              val userlandFlow = handler.handle()
-              eventStreamSource.via(userlandFlow)
-          }
+          scaladsl.Source
+            .fromFuture(futureOffset)
+            .initialTimeout(config.offsetTimeout)
```

Comment:
If `futureOffset` never completes (because of any underlying bug) that `Source` will never start and therefore the `RestartSource` wrapping it will never restart. 

The call to `initialTimeout` make sure that the `Source` fails if a first element (the offset) is not pushed throught it within the passed `offsetTimeout` (defaults to 5s)